server/image_pull.go

```diff
@@ -263,9 +266,24 @@ func (s *Server) pullImageCandidate(ctx context.Context, sourceCtx *imageTypes.S
 	return nil
 }
 
-// metricsFromProgressGoroutine consumes progress and turns it into metrics updates.
-func metricsFromProgressGoroutine(ctx context.Context, progress <-chan imageTypes.ProgressProperties, remoteCandidateName storage.RegistryImageReference) {
+// progressGoRoutine consumes progress and turns it into metrics updates.
+// It also checks if progress is being made within a constant timeout.
+// If the timeout is reached because no progress updates have been made, then
+// the cancel function will be called.
+func progressGoRoutine(ctx context.Context, cancel context.CancelFunc, progress <-chan imageTypes.ProgressProperties, remoteCandidateName storage.RegistryImageReference) {
```

Comment:
nit: maybe `consumeImagePullProgress` or something?

Reply:
Sure, following-up in https://github.com/cri-o/cri-o/pull/7946